controllers/permissionController.js

```diff
@@ -0,0 +1,147 @@
+import models from "../models";
+import constants from "../helpers/constants";
+import { isString, isEmpty } from "../helpers/funcValidators";
+
+const {
+  OK,
+  INTERNAL_SERVER_ERROR,
+  CONFLICT,
+  BAD_REQUEST,
+  CREATED
+} = constants.statusCode;
+const { Permission } = models;
+class PermissionController {
+  static async grantPermission(req, res) {
+    try {
+      const {
+        resource,
+        createPermission,
+        readPermission,
+        updatePermission,
+        deletePermission
+      } = req.body;
+      const { roleId } = req.params;
+      if (isEmpty(resource)) {
+        return res.status(BAD_REQUEST).json({
+          message: `Resource is required`
+        });
+      }
+      if (!isString(resource)) {
+        return res.status(BAD_REQUEST).json({
+          message: `{${resource}} can only be alphabetic characters`
+        });
+      }
+      const permission = await Permission.create({
+        resource,
+        roleId,
+        createPermission,
+        readPermission,
+        updatePermission,
+        deletePermission
+      });
+      return res.status(CREATED).json({
+        message: "Permission granted",
+        permission
+      });
+    } catch (error) {
+      if (error.name === "SequelizeUniqueConstraintError") {
+        return res.status(CONFLICT).json({
+          message: "Permission already exist"
+        });
+      }
+      return res.status(INTERNAL_SERVER_ERROR).json({
+        message:
+          "Sorry, this is not working properly. We now know about this mistake and are working to fix it"
+      });
+    }
+  }
+
+  static async getOnePermission(req, res) {
+    try {
+      const { permissionId } = req.params;
+      const permission = await Permission.findOne({
+        where: { id: permissionId }
+      });
+      return res.status(OK).json({
+        permission
```

Comment:
@niomwungeri-fabrice You should also consider the scenario when the permission is not found by returning `Permission not found` message

Reply:
@espoirMur there's a middleware to check that